src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -296,12 +300,49 @@ public static IHostBuilder AddScriptHostCore(this IHostBuilder builder, ScriptAp
                     services.TryAddEnumerable(ServiceDescriptor.Singleton<IHostedService, FunctionsScaleMonitorService>());
                 }
                 services.TryAddSingleton<FunctionsScaleManager>();
+
+                services.AddHostOverrides();
             });
 
             RegisterFileProvisioningService(builder);
             return builder;
         }
 
+        public static void AddHostOverrides(this IServiceCollection services)
+        {
+            // Custom implementations that the Host overrides
+            services.AddSingleton<ScheduleMonitor, AzureStorageScheduleMonitor>();
+            services.AddSingleton<IDistributedLockManager>(provider => AddLockManager(provider));
+        }
+
+        public static IDistributedLockManager AddLockManager(IServiceProvider provider)
+        {
+            try
+            {
+                var azureStorageProvider = provider.GetRequiredService<IAzureStorageProvider>();
+                var loggerFactory = provider.GetRequiredService<ILoggerFactory>();
+                var container = azureStorageProvider.GetBlobContainerClient();
+
+                if (container != null)
+                {
+                    return new BlobLeaseDistributedLockManager(loggerFactory, azureStorageProvider);
+                }
+            }
+            catch (Exception ex)
+            {
+                var str = ex.ToString();
```

Comment:
I don't understand this exception handler, it looks like it will swallow anything and no logs will be emitted

Reply:
Leftover from debugging :)